MapboxNavigation/InstructionsCardContainerView.swift

```diff
@@ -0,0 +1,315 @@
+            view.layer.insertSublayer(gradientLayer, at: 0)
+        }
+        
+        if let gradientLayer = view.layer.sublayers?.first as? CAGradientLayer {
+            gradientLayer.frame = bounds
+            gradientLayer.colors = colors
+        }
+        
+        if let nextBannerView = view as? NextBannerView {
+            nextBannerView.maneuverView.primaryColor = style.nextBannerViewPrimaryColor
+            nextBannerView.maneuverView.secondaryColor = style.nextBannerViewSecondaryColor
+            nextBannerView.instructionLabel.normalTextColor = style.nextBannerInstructionLabelTextColor
+            nextBannerView.instructionLabel.normalFont = style.nextBannerInstructionLabelNormalFont
+            nextBannerView.instructionLabel.shieldHeight = style.nextBannerInstructionLabelNormalFont.pointSize
+        }
+        
+        if let lanesView = view as? LanesView, let stackView = lanesView.subviews.first as? UIStackView {
+            let laneViews: [LaneView] = stackView.subviews.compactMap { $0 as? LaneView }
+            laneViews.forEach { laneView in
+                guard laneView.isValid else { return }
+                laneView.primaryColor = self.style.lanesViewDefaultColor
+                laneView.secondaryColor = self.style.lanesViewDefaultColor
+            }
+        }
+        
+        view.layoutIfNeeded()
+        
+        return view
+    }
+    
+    override public func layoutSubviews() {
+        super.layoutSubviews()
+    }
+    
+    private func gradientLayer(for view: UIView, with colors:[CGColor]? = nil) -> CAGradientLayer? {
+        guard !view.isHidden, let sublayers = view.layer.sublayers,
+              let firstLayer = sublayers.first as? CAGradientLayer,
+              let layerColors = firstLayer.colors as? [CGColor], layerColors.count == 2 else {
+            return nil
+        }
+        
+        if let colors = colors {
+            let colorsMatched = layerColors.reduce(false) { $0 || colors.contains($1) }
+            return colorsMatched ? firstLayer : nil
+        }
+        
+        return firstLayer
+    }
+    
+    public func updateInstruction(for step: RouteStep, distance: CLLocationDistance) {
+        instructionsCardView.updateDistanceFromCurrentLocation(distance)
+        instructionsCardView.step = step
+        
+        // TODO: Merge Instructions Card, Lanes & Next Banner View Instructions
+        guard let instruction = step.instructionsDisplayedAlongStep?.last else { return }
+        updateInstruction(instruction)
+        updateInstructionCard(distance: distance)
+    }
+    
+    public func updateInstruction(_ instruction: VisualInstructionBanner) {
+        if lanesView.isHidden {
+            lanesView.update(for: instruction)
+        } else if lanesView.isCurrentlyVisible && instruction.tertiaryInstruction == nil {
+            lanesView.hide()
+        }
+        
+        if nextBannerView.isHidden {
+            nextBannerView.update(for: instruction)
```

Comment:
I don’t think this update should be guarded by a conditional here. `NextBannerView.update(for:)` already hides the banner if there’s no information to display. But this conditional means a next banner only updates if it was hidden to begin with. I’m not totally sure that’s always correct.